packages/lisk-transactions/src/transactions/0_transfer.ts

```diff
@@ -0,0 +1,480 @@
+		const attr = super.getRequiredAttributes();
+
+		return {
+			[ENTITY_ACCOUNT]: {
+				address: [...attr[ENTITY_ACCOUNT].address, this.recipientId],
+			},
+		};
+	}
+
+	public processRequiredState(state: EntityMap): RequiredTransferState {
+		const { sender } = super.processRequiredState(state);
+		const accounts = state[ENTITY_ACCOUNT];
+		if (!accounts) {
+			throw new Error('Entity account is required.');
+		}
+		if (!isTypedObjectArrayWithKeys<Account>(accounts, ['address','publicKey'])) {
+			throw new Error('Required state does not have valid account type');
+		}
+	
+		const recipient = accounts.find(account => account.address === this.recipientId);
+		if (!recipient) {
+			throw new Error('No recipient account is found.');
+		}
+
+		return {
+			sender,
+			recipient,
+		};
+	}
+
+	public validateSchema(): TransactionResponse {
+		const { status, errors: baseErrors } = super.validateSchema();
+		const valid = validator.validate(transferAssetFormatSchema, this.asset);
+		const errors = [...baseErrors];
+		const assetErrors = validator.errors
+			? validator.errors.map(
+					error =>
+						new TransactionError(
+							`'${error.dataPath}' ${error.message}`,
+							this.id,
+							error.dataPath,
+						),
+			  )
+			: [];
+
+		errors.push(...assetErrors);
+
+		if (!validateTransferAmount(this.amount.toString())) {
+			errors.push(
+				new TransactionError(
+					'Amount must be a valid number in string format.',
+					this.id,
+					'.recipientId',
+				),
+			);
+		}
+
+		if (!this.recipientId) {
+			errors.push(
+				new TransactionError(
+					'`recipientId` must be provided.',
+					this.id,
+					'.recipientId',
+				),
+			);
+		}
+
+		try {
+			validateAddress(this.recipientId);
+		} catch (error) {
+			errors.push(new TransactionError(error.message, this.id, '.recipientId'));
+		}
+
+		if (
+			this.recipientPublicKey &&
+			this.recipientId !== getAddressFromPublicKey(this.recipientPublicKey)
+		) {
+			errors.push(
+				new TransactionError(
+					'recipientId does not match recipientPublicKey.',
+					this.id,
+					'.recipientId',
+				),
+			);
+		}
+
+		return {
+			id: this.id,
+			status:
+				status === Status.OK && valid && errors.length === 0
+					? Status.OK
+					: Status.FAIL,
+			errors,
+		};
+	}
+
+	public verify({
+		sender,
+		recipient,
+	}: RequiredTransferState): TransactionResponse {
+		const { errors: baseErrors } = super.apply({ sender });
+		const errors = [...baseErrors];
+		// Balance verification
+		const updatedSenderBalance = new BigNum(sender.balance).sub(this.amount);
+
+		if (!updatedSenderBalance.gte(0)) {
+			errors.push(
+				new TransactionError(
+					`Account does not have enough LSK: ${sender.address}, balance: ${
+						sender.balance
+					}`,
+					this.id,
+				),
+			);
+		}
+
+		const updatedRecipientBalance = new BigNum(recipient.balance).add(
+			this.amount,
+		);
+
+		if (!updatedRecipientBalance.lte(MAX_TRANSACTION_AMOUNT)) {
```

Comment:
im not sure this would be possible, but probably better to have?

Reply:
yes in core I believe it's being checked. would be good to check in case of errors.